clang/lib/Headers/cheri_init_globals.h

```diff
@@ -169,12 +169,6 @@ cheri_init_globals_impl(const struct capreloc *start_relocs,
     const void *__capability *__capability dest =
         (const void *__capability *__capability)cheri_address_or_offset_set(
             data_cap, reloc->capability_location + base_addr);
-    if (reloc->object == 0) {
-      /* XXXAR: clang fills uninitialized capabilities with 0xcacaca..., so we
```

Comment:
I wonder if we should also remove this from the .o files? It was great for debugging but I doubt it's needed anymore. But that can be a separate change.

Reply:
(c.f. https://git.morello-project.org/morello/llvm-project/-/blob/743973c6fa878c9d3bf003efe9baa4b7180d1645/lld/ELF/Arch/Cheri.cpp#L373-397; it used to do dodgy things instead by putting metadata there for the linker)